[New board] XIAO nRF52840

Hey,

I just wonder are the RNode Fw support XIAO nRF52840 & Wio-SX1262 board kit? They are very cheap and power efficient. I think the board hardware is the same as the RAK wisblock, only pin numbers are different. The hardware form factor is very tiny. 
https://www.seeedstudio.com/XIAO-nRF52840-Wio-SX1262-Kit-for-Meshtastic-p-6400.html?srsltid=AfmBOopVNvLuJn7kEwomoaxFggZI7D50XkzdmYcq8REHKF2-XWM85UO6

Thanks! 

Regards

testing my board.h right now, will add a pull request if i finish some nice working code

edit: firmware build, flashed, tmrw flash another xiaonrf and try meshchat to see if it works. after that i could share the progress i have made. wasnt to hard at all!

Here to proof of the code i wrote:

```
make upload-xiao_NRF
arduino-cli upload -p /dev/ttyACM0 --fqbn Seeeduino:nrf52:xiaonRF52840
Upgrading target on /dev/ttyACM0 with DFU package /home/admin/.cache/arduino/sketches/F7B83867168A0793978377B75CA0E829/RNode_Firmware_CE.ino.zip. Flow control is disabled, Single bank, Touch disabled
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
######################
Activating new firmware
Device programmed.
New upload port: /dev/ttyACM0 (serial)
unzip -o build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.zip -d build/Seeeduino.nrf52.xiaonRF52840
Archive:  build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.zip
 extracting: build/Seeeduino.nrf52.xiaonRF52840/manifest.json  
 extracting: build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.dat  
 extracting: build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.bin  
rnodeconf /dev/ttyACM0 --firmware-hash $(sha256sum ./build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.bin | grep -o '^\S*')
[02:01:03] Opening serial port /dev/ttyACM0...
[02:01:06] Device connected
[02:01:06] Current firmware version: 1.75
[02:01:06] Reading EEPROM...
[02:01:07] This device has not been provisioned yet, cannot set firmware hash
make: *** [Makefile:266: upload-xiao_NRF] Error 77

```

```
rnodeconf /dev/ttyACM0 -r --platform NRF52 --model ff --product f0 --hwrev 1
[02:03:21] Opening serial port /dev/ttyACM0...
[02:03:24] Device connected
[02:03:24] Current firmware version: 1.75
[02:03:24] Reading EEPROM...
[02:03:25] Waiting for NRF52 reset...
[02:03:43] Loading signing key...
[02:03:43] Bootstrapping device EEPROM...
[02:04:15] EEPROM written! Validating...
[02:04:17] Waiting for NRF52 reset...
[02:04:38] EEPROM checksum correct
[02:04:38] Device signature validated
[02:04:38] EEPROM Bootstrapping successful!
```
after this com changed randomly to acm1

```
`arduino-cli upload -p /dev/ttyACM1 --fqbn Seeeduino:nrf52:xiaonRF52840
Upgrading target on /dev/ttyACM0 with DFU package /home/admin/.cache/arduino/sketches/F7B83867168A0793978377B75CA0E829/RNode_Firmware_CE.ino.zip. Flow control is disabled, Single bank, Touch disabled
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
########################################
######################
Activating new firmware
Device programmed.
New upload port: /dev/ttyACM0 (serial)
unzip -o build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.zip -d build/Seeeduino.nrf52.xiaonRF52840
Archive:  build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.zip
 extracting: build/Seeeduino.nrf52.xiaonRF52840/manifest.json  
 extracting: build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.dat  
 extracting: build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.bin  
rnodeconf /dev/ttyACM0 --firmware-hash $(sha256sum ./build/Seeeduino.nrf52.xiaonRF52840/RNode_Firmware_CE.ino.bin | grep -o '^\S*')
[02:07:06] Opening serial port /dev/ttyACM0...
[02:07:09] Device connected
[02:07:09] Current firmware version: 1.75
[02:07:09] Reading EEPROM...
[02:07:10] EEPROM checksum correct
[02:07:10] Device signature validated
[02:07:10] Firmware hash set``
```

--- a/Makefile
+++ b/Makefile
@@ -51,6 +51,7 @@ prep-nrf:
 	arduino-cli core install adafruit:nrf52 --config-file arduino-cli.yaml
 	arduino-cli core install rakwireless:nrf52 --config-file arduino-cli.yaml
 	arduino-cli core install Heltec_nRF52:Heltec_nRF52 --config-file arduino-cli.yaml
+	arduino-cli core install Seeeduino:nrf52 --config-file arduino-cli.yaml
 	arduino-cli lib install "Crypto"
 	arduino-cli lib install "Adafruit GFX Library"
 	arduino-cli lib install "GxEPD2"

--- a/arduino-cli.yaml
+++ b/arduino-cli.yaml
@@ -4,3 +4,4 @@ board_manager:
   - https://liberatedsystems.co.uk/rnode-firmware-ce/esp-custom-package.json
   - https://raw.githubusercontent.com/RAKwireless/RAKwireless-Arduino-BSP-Index/main/package_rakwireless_index.json
   - https://github.com/HelTecAutomation/Heltec_nRF52/releases/download/1.7.0/package_heltec_nrf_index.json
+  - https://files.seeedstudio.com/arduino/package_seeeduino_boards_index.json

